yb_migrate/src/srcdb/srcdb.go

```diff
@@ -14,6 +16,7 @@ type SourceDB interface {
 	GetAllPartitionNames(tableName string) []string
 	ExportSchema(exportDir string)
 	ExportData(ctx context.Context, exportDir string, tableList []string, quitChan chan bool, exportDataStart chan bool)
+	ExportDataPostProcessing(source *Source, exportDir string, tablesProgressMetadata *map[string]*utils.TableProgressMetadata)
```

Comment:
This needs more code re-arrangement. The `ExportDataPostProcessing` is equivalent to `saveExportedRowCount()` for Oracle and MySQL. For PG, it is equivalent of `renameDataFiles() + saveExportedRowCount()`. If you take a closer look, the `renameDataFiles()` is related to `ExportData()` whereas `saveExportedRowCount()` is more a part of migration state keeping. In fact, the `saveExportedRowCount()` doesn't interact with source db at all--hence it should not be part of SourceDB. Here is a better alternative:

-- Move the `renameDataFiles()` functionality as the last step in the  `Postgres.ExportData()`.
-- That leaves you with only the `saveExportedRowCount()` in the `ExportDataPostProcessing()`. Move the call to the `saveExportedRowCount()` in the `exportDataOffline()` where the earlier call to `ExportDataPostProcessing()` was made.
-- Get rid of the `ExportDataPostProcessing()` function.